pkg/kubectl/cmd/describe.go

```diff
@@ -59,7 +59,7 @@ $ kubectl describe po -l name=myLabel`,
 
 func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
 	selector := cmdutil.GetFlagString(cmd, "selector")
-	cmdNamespace, err := f.DefaultNamespace()
+	cmdNamespace, _, err := f.DefaultNamespace()
```

Comment:
Why don't we do the overridden check in describe?

Describe and get commands can take resource descriptions from files (which could specify namespace) and use those to retrieve the corresponding current state.